### PR TITLE
Add material-date-picker

### LINKED ISSUE #11: Material date-picker
- [Visual specs](https://drive.google.com/open?id=0BzNRYMf7bO7kdFhDTXlUTGhzSHM) ([Sketch source file](https://drive.google.com/open?id=0BzNRYMf7bO7kaTJJUmV4OXJwcUk))

### PR DESCRIPTION
**Pull request by Saulis**
Fixes #11

I'll be providing visual regression tests in a separate PR from this. This review should focus on the theme only.

---
This change is [Reviewable](https://reviewable.io/reviews/vaadin/vaadin-themes/15)

### CODE REVIEW

**Comment by jouni** [comment]
Someone’s working late ;)

**Comment by jouni** [comment]
Dark theme is broken, the month scroller background is white for me (Chrome and Safari).

File: material/elements/vaadin-date-picker.html [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+        color: var(--material-primary-contrast-color);
+        background: transparent;
+        box-sizing: border-box;
+      }
+
+      [part~="date"][selected] {
+        font-weight: 600;
+      }
+
+      [part~="weekdays"] {
```
Comment (jouni) [inline material/elements/vaadin-date-picker.html]: Isn’t this selector unnecessary? Move the font size property (and use the correct value, not `small`) to the following selector (weekday) and remove the color, as it’s overridden by the next selector.

File: material/elements/vaadin-date-picker.html [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+        color: rgba(0, 0, 0, 0.5);
+        font-size: small;
+      }
+
+      [part~="weekday"] {
+        color: var(--material-secondary-text-color);
+      }
+
+      [part~="week-number"] {
+        height: 32px;
```
Comment (jouni) [inline material/elements/vaadin-date-picker.html]: Should be the same as date cell (`calc(var(--material-base-grid) * 5)`).

**Comment by jouni** [comment]
General observation: I think there’s usually no need to use to `~=` attribute selector in the themes, as most parts only contain a single name (i.e. `part="foo"` vs. `part="foo bar"`). We can save a few bytes by just using `[part="foo"]` (and not `[part~="foo"]`).

File: material/elements/vaadin-date-picker.html [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+        position: absolute;
+        top: 0;
+        left: 0;
+        width: 4px;
+        height: 100%;
+        background: linear-gradient(.25turn, rgba(0, 0, 0, 0.1), rgba(0, 0, 0, 0));
+      }
+
+      [part~="months"] {
+        --vaadin-infinite-scroller-item-height: 300px;
```
Comment (jouni) [inline material/elements/vaadin-date-picker.html]: This should be based on `--material-base-grid`. Perhaps we need to fix the height of the `month-header` and `weekdays` elements using that as well?

File: material/elements/vaadin-date-picker.html [hunk 4]
```diff
@@ -0,0 +1,218 @@
...
+      [part~="toolbar"] {
+        padding: 4px;
+        color: var(--material-body-text-color);
+        background: var(--material-secondary-background-color);
+        border-top: 1px solid rgba(0, 0, 0, 0.3);
+      }
+
+      [part~="overlay-header"] {
+        color: var(--material-body-text-color);
+        font-size: 18px;
```
Comment (jouni) [inline material/elements/vaadin-date-picker.html]: Font size should be the same as for the normal version. No need to increase.

**Comment by jouni** [comment]
### Fullscreen mode

The fullscreen mode is missing the text field label. It should be visible there as well. The header should basically visually contain a focused vaadin-text-field, where the underline spans the entire viewport width, while the other parts have a small padding to the viewport edges (--material-space-s).

The year toggle button is missing border-radius from the bottom left corner. It could be a tab taller as well (make it 40px), to accommodate taps easier.

**Comment by jouni** [comment]
I think we could next time try using zeplin.io so it will be easier to pick up the specs from the Sketch file.

**Comment by limonte** [comment]
:lgtm:

---

Reviewed 4 of 6 files at r1, 3 of 6 files at r2, 2 of 7 files at r3, 1 of 1 files at r4, 20 of 20 files at r5.
Review status: all files reviewed at latest revision, 4 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-themes/15#-:-KmkhkZk7PQco59fF1m4:bnfp4nl)*

**Pull request merged**